### PR TITLE
net: lwm2m: Add ObjLnk support

### PR DESCRIPTION
**Pull request by rlubos**
Add support for ObjLnk resource type, and implement it for ExtDevInfo resource in Device object. Additionally, fix a few other issues that showed up during tests.

I was able to verify plaintext and TLV encoding with Leshan successfully. As for JSON, it turned out that ObjLnk JSON encoding is not implemented in Leshan, so it keeps rejecting this resource when JSON is used. I've inspected the packet manually with Wireshark, it looked fine and Wireshark had no issues decoding it correctly.

### CODE REVIEW

File: subsys/net/lib/lwm2m/Kconfig [hunk 1]
```diff
@@ -240,6 +240,14 @@ config LWM2M_DEVICE_ERROR_CODE_MAX
 	  This value sets the maximum number of error codes that the device
 	  object will store before ignoring new values.
 
+config LWM2M_DEVICE_EXT_DEV_INFO_MAX
+	int "Maximum # of device obj external device info to store"
+	default 5
+	range 1 20
```
Comment (jukkar) [inline subsys/net/lib/lwm2m/Kconfig:246]: Is the range coming from some spec, or are the values here just selected to be suitable common values?

**Review by jukkar — APPROVED** [review]
LGTM

File: subsys/net/lib/lwm2m/Kconfig (hunk 1 above)
Comment (rlubos) [reply subsys/net/lib/lwm2m/Kconfig:246]: Spec does not define maximum number of multiinstance resources, I've just used some arbitrary values as other multiinstance resources above.

**Review by tbursztyka — APPROVED** [review]

**Comment by jukkar** [comment]
@rlubos could you fix the merge conflict

**Comment by rlubos** [comment]
@jukkar Sorry for the delay, should be fixed now.

**Pull request merged**